src/network/connection.ts

```diff
@@ -0,0 +1,941 @@
+import { markOnce } from '@src/lib/performance'
+import type { IEventListenerTracked, ManagerTearDown } from '@src/network/utils'
+import {
+  ConnectingType,
+  DATACHANNEL_NAME_UMC,
+  EngineConnectionEvents,
+  EngineConnectionStateType,
+  pingIntervalMs,
+  promiseFactory,
+  WebSocketStatusCodes,
+} from '@src/network/utils'
+import {
+  createOnConnectionStateChange,
+  createOnDataChannel,
+  createOnIceCandidate,
+  createOnIceCandidateError,
+  createOnIceConnectionStateChange,
+  createOnIceGatheringStateChange,
+  createOnNegotiationNeeded,
+  createOnSignalingStateChange,
+  createOnTrack,
+} from '@src/network/peerConnection'
+import {
+  createOnWebSocketClose,
+  createOnWebSocketError,
+  createOnWebSocketMessage,
+  createOnWebSocketOpen,
+} from '@src/network/websocketConnection'
+import { EngineDebugger } from '@src/lib/debugger'
+import { uuidv4 } from '@src/lib/utils'
+import { withWebSocketURL } from '@src/lib/withBaseURL'
+import type {
+  WebSocketRequest,
+  WebSocketResponse,
+  ClientMetrics,
+} from '@kittycad/lib/dist/types/src'
+
+// An interface for a promise that needs to be awaited and pass the resolve reject to
+// other dependencies. We do not need to pass values between these. It is mainly
+// to know did it pass or fail, not what the result is?
+// A specific case from the promiseFactor<T>() call
```

Comment:
I think there's a typo, and I'm not sure why this comment is phrased as a question.

```suggestion
// to know did it pass or fail, not what the result is.
// A specific case from the promiseFactory<T>() call
```

Reply:
handled.